tsdb/engine/tsm1/reader.go

```diff
@@ -235,9 +235,11 @@ func NewTSMReader(f *os.File, options ...tsmReaderOption) (*TSMReader, error) {
 		f:            f,
 		mmapWillNeed: t.madviseWillNeed,
 	}
+	t.accessor.close()
```

Comment:
What is the reason for this `close()` here?

Reply:
I added this line to check if it does fix the failing tests on Windows by trial and error. I can't remember which exact tests I was trying to fix in this case, but it was definitely related to the error that `t.TempDir()` fails to cleanup the temporary directory because a file is still opened.

The CI is passing so when this line is removed so it is a leftover in the process of trial and error. The actual fix is the addidion of :arrow_down: 

```diff
	index, err := t.accessor.init()
	if err != nil {
+		_ = t.accessor.close()
		return nil, err
	}
```

---

https://app.circleci.com/pipelines/github/influxdata/influxdb/27855/workflows/e3d86254-abb2-41db-b339-c03b04248b41/jobs/250050  <-- This is one of the early pipeline run. Many tests were failing on Windows because we were not doing cleanups properly in many of the tests. Some of the tests were easy to fix by doing e.g. closing an open file in the test, while some of them requires the cleanup to be done on the source code